services/basic_types.proto

```diff
@@ -1592,3 +1593,32 @@ message StakingInfo {
         int64 staked_node_id = 6;
     }
 }
+
+message VirtualAddress {
+    /**
+     * The 20-byte EVM address.
+     */
+    bytes address = 1;
+
+    /**
+     * The ethereum transaction nonce associated with this address.
+     */
+    int64 nonce = 2;
```

Comment:
I'm not convinced we need a `nonce` field at all. 
For a non-contract account's EVM address, the nonce can only be updated by one for each `EthereumTransaction` initiated by this EVM address. The mirror node already has logic to increment the nonce by one in this case. It only needs to be updated and increment the nonce field of the appropriate EVM address belonging to this account.
We currently don't update the nonces of contract accounts and there is a different approach discussed for contract account nonces anyway (reference: https://github.com/hashgraph/hedera-protobufs/issues/245)

Reply:
I agree that incrementing an account's nonce only when executing `EthereumTransactions` may be a bit confusing. But it would require very careful consideration if we were to start updating the nonce with each HAPI ContractCall and ContractCreate transaction. Some things we would need to consider:
- If we do start updating the nonce field, then we should verify it always has a correct value in the signed transaction body. But we don't need to do this, Hedera has a different mechanism for protection against replay attacks.
- In Ethereum any transaction would increase the nonce, even a simple transfer, so why not include `CryptoTransfer` transactions as well, and then, why not all other types of transactions as well, like TokenMint? 
- Some external tools that work with the Relay have their own logic for tracking account nonces, and having Hedera increase the nonce value for this account address for all HAPI transactions submitted by this account may just confuse these tools and make them harder to integrate with Hedera
- What about synthetic child transactions, should they increase the nonce of the accounts? Would that mean that one HAPI transaction can increase an account nonce several times (if for example multiple calls to the precompiled contracts are executed)

Reply:
>If we do start updating the nonce field, then we should verify it always has a correct value in the signed transaction body. But we don't need to do this, Hedera has a different mechanism for protection against replay attacks.

`EthereumTransaction` does this.  ContratCallTransaction and all other transactions don't specify a nonce.

> In Ethereum any transaction would increase the nonce, even a simple transfer, so why not include CryptoTransfer transactions as well, and then, why not all other types of transactions as well, like TokenMint?

Ethereum nonce management is a nightmare, we don't want to put that upon ourselves.  We only have nonces in `EthereumTransaction` for semantic compatibility, otherwise we would not track it.

> Some external tools that work with the Relay have their own logic for tracking account nonces, and having Hedera increase the nonce value for this account address for all HAPI transactions submitted by this account may just confuse these tools and make them harder to integrate with Hedera

Which is why it was limited to `EthereumTransaction` - the act of signing a tx with that nonce is where the tooling should track and manage nonces.

> What about synthetic child transactions, should they increase the nonce of the accounts? Would that mean that one HAPI transaction can increase an account nonce several times (if for example multiple calls to the precompiled contracts are executed)

Another reason to only track for top level `EthereumTransaction` objects.

Reply:
I'm persuaded by the above collective sentiments.
The hope was we'd cover the nonce for EVM interacting transactions only.
However, seems the cons outweigh the pros at this juncture.